### PR TITLE
Admin frontend improvements

### PR DESCRIPTION
**Pull request by bratseth1**
Review and import whatever you need from this branch at this point. The work remaining in this branch could be done on a new branch.  Will continue the work from 11.05.24 as we discussed.

### CODE REVIEW

**Comment by SmileyFace799** [comment]
Gonna review this, kind of a humongous commit so will take a while, try to pull request a bit sooner next time

File: frontend/package.json [hunk 1]
```diff
@@ -11,10 +11,11 @@
   "dependencies": {
     "@mdi/font": "^7.4.47",
     "@mdi/js": "^7.4.47",
-    "axios": "^1.6.7",
+    "axios": "^1.6.8",
     "chart.js": "^4.4.2",
     "swiper": "^11.1.1",
     "vue": "^3.4.15",
+    "vue-jwt-decode": "^0.1.0",
```
Comment (SmileyFace799) [inline frontend/package.json]: This seems to be a 6 year old library for vue 2, use a different library that works with vue 3

File: backend/src/main/java/no/ntnu/database/controllers/TwoFactorRequests.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import no.ntnu.security.SecurityUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * Service for handling two-factor authentication operations.
+ */
+@Service("twoFactorRequests")
+public class TwoFactorRequests {
```
Comment (SmileyFace799) [inline backend/src/main/java/no/ntnu/database/controllers/TwoFactorRequests.java]: This should be a service class inside the `services` package

File: frontend/package-lock.json [hunk 3]
```diff
@@ -1720,6 +1744,41 @@
         }
       }
     },
+    "node_modules/vue-jwt-decode": {
+      "version": "0.1.0",
+      "resolved": "https://registry.npmjs.org/vue-jwt-decode/-/vue-jwt-decode-0.1.0.tgz",
+      "integrity": "sha512-4iP0NzYHkAF7G13tYPc/nudk4oNpB8GCVZupc7lekxXok1XKEgefNaGTpDT14g7RKe5H9GaMphPduDj4UVfZwQ==",
+      "dependencies": {
+        "vue": "^2.3.3"
```
Comment (SmileyFace799) [inline frontend/package-lock.json]: Depends on vue 2, bad

File: frontend/package.json (hunk 1 above)
Comment (bratseth1) [reply frontend/package.json]: Fixed

File: frontend/package-lock.json (hunk 3 above)
Comment (bratseth1) [reply frontend/package-lock.json]: Fixed

File: backend/src/main/java/no/ntnu/database/controllers/TwoFactorRequests.java (hunk 2 above)
Comment (bratseth1) [reply backend/src/main/java/no/ntnu/database/controllers/TwoFactorRequests.java]: This was moved temporarily from service package for testing purpose. Has been moved back.

**Review by SmileyFace799 — CHANGES_REQUESTED** [review]
This breaks the search bar, as access to the /course & /provider endpoints are restricted to logged in users. This must be fixed before merge

**Review by SmileyFace799 — APPROVED** [review]
Merge shouldn't cause any issues with main now, will merge this & hope it works 🙏

**Pull request merged**